scripts/os_configure/install_software_20.bash

```diff
@@ -1,84 +1,172 @@
+  echo -e "${TERMINAL_COLOR_USER_INPUT_DECISION} office   -Used for office pcs. Basic utilities, additional tools and office related stuff."
+  echo -e "${TERMINAL_COLOR_USER_INPUT_DECISION} robot    -Used for robot platforms. Basic utilities like development utilities (git, vscode, ...)."
+  select computer_type in basic office robot;
+  do
+    case "$computer_type" in
+          basic)
+              computer_type="basic"
+              break
+            ;;
+          office)
+              computer_type="office"
+              break
+            ;;
+          robot)
+              computer_type="robot"
+              break
+            ;;
+    esac
+  done
+  echo -n -e "${TERMINAL_COLOR_NC}"
 fi
 
 echo "Installing software for $computer_type computers. Press <ENTER> to continue."
 read
 
 ROS_VERSION=melodic
-ROS2_VERSION=foxy
+ROS2_VERSIONS=( "galactc" "rolling" )
 
 # Core tools
-sudo apt update && sudo apt -y install curl gnupg2 lsb-release
+sudo apt update && sudo apt -y install ca-certificates curl gnupg2 lsb-release
 
-#KDE Backports
+# KDE Backports
 sudo apt-add-repository -y ppa:kubuntu-ppa/backports
 sudo apt update && sudo apt -y dist-upgrade && sudo apt -y autoremove
 
-### Useful tools
-sudo apt -y install vim ssh git trash-cli htop unrar yakuake screen finger ksshaskpass kompare filelight
+# Nala - better apt frontend
+echo "deb http://deb.volian.org/volian/ scar main" | sudo tee /etc/apt/sources.list.d/volian-archive-scar-unstable.list
+wget -qO - https://deb.volian.org/volian/scar.key | sudo tee /etc/apt/trusted.gpg.d/volian-archive-scar-unstable.gpg > /dev/null
+sudo apt update
+sudo apt -y install nala
+nala --install-completion bash
+
 # Dolphin Plugins
 sudo apt -y install kdesdk-kio-plugins kdesdk-scripts
 
-if ([[ $computer_type != "robot" ]])
-then
-  sudo apt -y install recordmydesktop rdesktop gimp meshlab inkscape pdfposter unrar
-  sudo DEBIAN_FRONTEND=noninteractive apt -y install wireshark
-#   sudo apt -y install gvfs-bin gvfs-fuse gvfs-backends
-fi
-
-# Useful libraries
-sudo apt -y install libxml2-dev libvlc-dev libmuparser-dev libudev-dev
-
-if ([[ $computer_type != "robot" ]])
-then
-  # Latex
-  sudo apt -y install kile texlive-full texlive-lang-german kbibtex ktikz
-fi
+## Useful tools
+sudo apt -y install neovim ssh git qgit trash-cli htop unrar yakuake screen finger ksshaskpass kompare filelight tldr thefuck ranger tree
 
 # Python tools
-sudo apt -y install python3-pip
+sudo apt -y install python3-pip \
+  python3-colcon-common-extensions \
+  python3-flake8 \
+  python3-flake8-blind-except \
+  python3-flake8-builtins \
+  python3-flake8-class-newline \
+  python3-flake8-comprehensions \
+  python3-flake8-deprecated \
+  python3-flake8-docstrings \
+  python3-flake8-import-order \
+  python3-flake8-quotes \
+  python3-pytest \
+  python3-pytest-cov \
+  python3-pytest-repeat \
+  python3-pytest-rerunfailures \
+  python3-rosdep \
+  python3-setuptools \
+  python3-vcstool
```

Comment:
Do we really need list each of those commands?

Reply:
not sure, seems that they don't come preinstalled and most of them are useful

Reply:
We could include them like i did with the [vs_code extenions](https://github.com/StoglRobotics/ros_team_workspace/pull/57/files#diff-6cdd466837a1cbec68d6261d18ac2c4ee821c343e4c4cccb6b7f3e904103abd3)